### ISSUE TITLE
Differents questions/difficulties with SSO

### ISSUE THREAD

**Issue by Michenux** [flytreeleft/nexus3-keycloak-plugin#31]
1) oidc configuration

In the oidc configuration, i had to put the IP so it can work -> set $target http://172.18.0.4:8081;
If i use container name, it tells me no resolver are configured. I had to put resolver 8.8.8.8; in nginx configuration so the error disappears but this dns doesnot know about my container names.

2) System status

In nexus, the system status does not work -> "You're trying to decode an invalid json".
In chrome developer tools, i can see these urls returns a 302 redirect to keycloak.

The invoked urls are :
https://nexus******/service/metrics/healthcheck?_dc=1572515536638
https://nexus******/static/rapture/baseapp-prod.js?_v=3.19.1-01&_e=OSS:1

If i copy/paste these urls in another tab, it works well.

3) settings.xml

How do you configure your settings.xml(maven) to pass SSO authentication ?
Possible ?

4) On my nexus Homepage only, a download dialog appears. Not on the other pages.

**Comment by flytreeleft** [comment]
- About `oidc configuration`

Just add `resolver 127.0.0.11 ipv6=off;` (details in [Docker Network Nginx Resolver](https://stackoverflow.com/questions/35744650/docker-network-nginx-resolver#answer-37656784)) to your configuration.

- About `System status`

I have checked the menu `Support -> Status` in my local environment, and it works well.

Can you see the status table like the following picutre?

![gtkshot](https://user-images.githubusercontent.com/1321315/68072079-b0004f00-fdbc-11e9-91cc-7a613dbed097.png)

If you can see it, maybe you can omit the error. If not, I think you should check your Nginx configuration carefully.

- About `settings.xml`

You need to omit your OIDC settings when the HTTP header `Authorization` isn't empty, so change your settings like:

```nginx
if ($http_authorization ~* "^$") {
    set $oidc_xxx "xxx";
    ...

    access_by_lua_file /path/to/oidc.lua;
}
```

- About `download dialog appears`

I've never encountered this problem. Can you provide a screenshot and the request/response header of the related url?

**Comment by Michenux** [comment]
**1) oid configuration** -> fixed, thank a lot.
**2) System configuration** -> fixed, forget to change $oidc_realm in nginx.

**3) settings.xml : not fixed.**

When i try in postman with a basic auth, it works well.
When i try with maven, the download is redirected to the keycloak login page.

My settings.xml : 
```
<mirrors>
  <mirror>
    <id>nexus</id>
    <mirrorOf>*</mirrorOf>
    <url>https://nexus.*****/repository/maven-public/</url>
  </mirror>
</mirrors>
<servers>
  <server>
    <id>nexus</id>
    <username>mylogin</username>
    <password>mypassword</password>
  </server>
</servers>
```


**4) Download: not fixed.**

The invoked url is :
https://nexus.****/service/outreach/?version=3.19.1-01&versionMm=3.19&edition=OSS&usertype=admin

![image](https://user-images.githubusercontent.com/650789/68115771-784af180-fef9-11e9-8233-bedccffe7cc7.png)

It comes from the Outreach plugin which retrieves data from sonatypeservice. I have disabled the plugin for the moment. 
https://help.sonatype.com/repomanager2/configuration/configuring-outreach-content-in-welcome-tab

**Comment by flytreeleft** [comment]
- About `settings.xml`

Just add the following statements to your Nginx configuration:

```nginx
        set $oidc_disabled "false";
        # Disable OIDC when using maven client
        if ($http_user_agent ~* "^(Apache-Maven|docker)/.+$") {
            set $oidc_disabled "true";
        }
        # And disable OIDC when the header Authorization was specified
        if ($http_authorization !~* "^$") {
            set $oidc_disabled "true";
        }
```

The bug has been fixed in commit [91962fc](https://github.com/flytreeleft/nexus3-keycloak-plugin/commit/91962fc90779d9348a55c71329f407481d52d836).

- About `Download`

I have confirmed that the Outreach plugin will indeed cause a download dialog to be shown, but I haven't found the reason, so just disable it until the resolution is provided.

**Comment by flytreeleft** [comment]
Close the inactive issue, if there still are some problems, please just make a comment or create a new issue :)
